File: PHP/Test/PackageStateCapture.php
<?php

namespace Test\PackageStateCapture;

use One\Ast\Types\Package;
use Utils\TSOverviewGenerator\TSOverviewGenerator;

class PackageStateCapture {
    public $overviews;
    public $pkg;
    
    function __construct($pkg) {
        $this->pkg = $pkg;
        $this->overviews = Array();
        foreach (array_values($pkg->files) as $file)
            $this->overviews[$file->sourcePath->path] = (new TSOverviewGenerator(false, false))->generate($file);
    }
    
    function getSummary() {
        return implode("\n\n", array_map(function ($file) { return "=== " . $file . " ===\n\n" . @$this->overviews[$file] ?? null; }, array_keys($this->overviews)));
    }
}

File: CSharp/Test/PackageStateCapture.cs
using One.Ast;
using System.Collections.Generic;
using Utils;

namespace Test
{
    public class PackageStateCapture {
        public Dictionary<string, string> overviews;
        public Package pkg;
        
        public PackageStateCapture(Package pkg)
        {
            this.pkg = pkg;
            this.overviews = new Dictionary<string, string> {};
            foreach (var file in Object.values(pkg.files))
                this.overviews.set(file.sourcePath.path, new TSOverviewGenerator(false, false).generate(file));
        }
        
        public string getSummary()
        {
            return Object.keys(this.overviews).map(file => $"=== {file} ===\n\n{this.overviews.get(file)}").join("\n\n");
        }
    }
}